Add description for out of bounds MBC ram access (#594)
gbdev/pandocs: src/MBCs.md

which use only code and data which is located in internal RAM. Despite the 
above, a self-made MBC1-EPROM card appears to work stable and fine even in 
Double Speed Mode.

## MBC Unmapped RAM Bank Access

> On most of the MBC's in case an unmapped RAM bank is selected (which would be translate to an out of bounds RAM address by the MBC controller), 

Reviewer: ```suggestion
In most MBCs, if an unmapped RAM bank is selected (which would be translate to an out of bounds RAM address by the MBC controller), 
```

> the MBC controller will simply wrap around the internal ram address and would access valid RAM address.

Reviewer: the repetition of "MBC controller" doesn't sound good here

> The MBC internal address being accessed can be calculated using this formula: `((address - external_ram_start_address) * ram_bank_size) % max_external_ram_size`.

Reviewer 2: I'd have expected `(active_ram_bank % nb_ram_banks) * 0x1000 + (address - 0xA000)`. Am I missing something?

Author: You are correct, I forgot to add the active ram bank to the formula.

I wrote the formula differently but they are supposed to be equivalent (unless I made a mistake again), let me know which way you prefer it.

which use only code and data which is located in internal RAM. Despite the 
above, a self-made MBC1-EPROM card appears to work stable and fine even in 
Double Speed Mode.

## MBC Out Of Bounds RAM Access

> On most of the MBC's in case an out of bounds address is accessed (selecting unmapped RAM bank),

Reviewer 2: Is there any MBC on which this doesn't happen?

Reviewer 3: In MBC3, banks $08-$0C select the RTC, and in MBC7, bank $40 selects the accelerometer and EEPROM.

Reviewer 4: What happens if you request banks $04-$07 or $0D-$0F?
I actually have everything you need to check this, I'm just lazy. :sloth:

Reviewer 5: $04-$07 and $0D-$0F just map to "nothing", read returning open bus and writes doing nothing. (also this is assuming MBC3, with MBC30 $04-$07 are just more RAM banks as expected)

Reviewer 4: I decided to not be lazy and tested this. $04-$07 and $0D-$0F return $FF on every address and are read-only. The high RAM bank nibble is ignored. So $4000 is effectively a 4-bit register.

Reviewer 5: The 0xFF is actually just an open bus value: 0xFF would just be a usual open bus value if you're reading from somewhere else other than the cartridge bus (e.g. WRAM within the GBC, or just HRAM in all units). It can be understood as just being the same as reading SRAM while SRAM is disabled via the 0000-1FFF register.

Reviewer 2: $FF is a common SRAM open-bus value especially with some flashcarts, so this makes sense.

Author: I believe others know better than me (as appears in this thread), since there are many types and shapes of MBC's I thought it could be a good idea to mention this is probably valid for most of them and not all of them.

> only the mapped bits from the address will be used.

Reviewer 2: Wait--is this talking about address, or bank? I know they are combined to ultimately make a single address passed to the chip; but the entire document is written from the perspective of the GB CPU, since that's the most consistent and generally-useful one, and from that perspective, the bank and address are separate.

Author: Agree! Just rephrased it to be more CPU oriented, let me know what you think
